### PR TITLE
added Ykush 3port usb hub

### PR DESCRIPTION
**Pull request by StefanRB**
Detailed changes:
- split existing class handling YkushXS into `YkushBase(PDUDriver)` and `YkushXS(YkushBase)`
- added `Ykush(YkushBase)` to handle 3-Port Ykush USB Hub
- renamed share/80-ykushxs.rules to share/80-ykush.rules and addes ykush 3-port handling to it

### CODE REVIEW

File: pdudaemon/drivers/ykush.py [hunk 1]
```diff
@@ -24,26 +25,27 @@
...
+    port_count = 0
 
     def __init__(self, hostname, settings):
         self.hostname = hostname
         self.settings = settings
         self.serial = settings.get("serial", u"")
         log.debug("serial: %s" % self.serial)
 
-        super(YkushXS, self).__init__()
+        super(YkushBase, self).__init__()
```
Comment (doanac) [inline pdudaemon/drivers/ykush.py]: Since its Python3 code, you can just do super().__init__() and copy/paste gets more fool-proof

Comment (StefanRB) [reply pdudaemon/drivers/ykush.py]: good hint, thanks! I'll wait for the review from @sjoerd-ccu and update

**Comment by mattface** [comment]
@sjoerd-ccu can you review this as it changes your driver.

@StefanRB thanks for the PR, can you do a simple rebase as I merged another driver earlier.

File: pdudaemon/drivers/ykush.py [hunk 2]
```diff
@@ -52,19 +54,37 @@ def port_interaction(self, command, port_number):
...
-        d.open(YKUSH_VID, YKUSH_XS_PID, serial_number=self.serial)
+        d.open(YKUSH_VID, self.ykush_pid, serial_number=self.serial)
         d.write([byte, byte])
         d.read(64)
         d.close()
 
     @classmethod
     def accepts(cls, drivername):
-        return drivername in cls.supported
+        return False
```
Comment (sjoerdsimons) [inline pdudaemon/drivers/ykush.py]: If you keep this one functional rather then dummy (doesn't really have to be a dummy here in the first place) then you would just have to set support = [..]. in the classes that inherent rather then reimplementing the accepts method.

I guess you could even make accepts just be `drivername == cls.__name__.upper()`

**Comment by StefanRB** [comment]
@mattface I've rebased and updated the driver according to the findings from @sjoerdsimons and @doanac

**Comment by mattface** [comment]
Thanks very much!

**Pull request merged**